File: backend/src/validators/templateValidatorRegistry.ts
import {
    validateAwsS3Request,
    type TemplateValidationInput,
    type TemplateValidationResult,
  } from "./s3RequestValidator.js";
  
  export type TemplateValidator = (
    input: TemplateValidationInput
  ) => TemplateValidationResult;
  
  const templateValidatorMap: Record<string, TemplateValidator> = {
    "aws-s3-bucket": validateAwsS3Request,
  };
  
  export function runTemplateValidator(
    templateId: string,
    input: TemplateValidationInput
  ): TemplateValidationResult {
    const validator = templateValidatorMap[templateId];
  
    if (!validator) {
      return {
        valid: true,
        errors: [],
        normalizedParameters: input.parameters,
      };
    }
  
    return validator(input);
  }

File: backend2/app/validators/template_validator_registry.py
from app.validators.s3_request_validator import validate_aws_s3_request

template_validator_map = {
    "aws-s3-bucket": validate_aws_s3_request,
}

def run_template_validator(template_id: str, input_data: dict) -> dict:
    validator = template_validator_map.get(template_id)
    if not validator:
        return {
            "valid": True,
            "errors": [],
            "normalizedParameters": input_data["parameters"],
        }

    return validator(input_data)